cmd/minishift/cmd/util/util.go

```diff
@@ -221,3 +223,29 @@ func DetermineClusterUpParameters(config *clusterup.ClusterUpConfig, Dockerbridg
 
 	return clusterUpParams
 }
+
+func EnsureConfigFileExists(configPath string) {
+	if _, err := os.Stat(configPath); os.IsNotExist(err) {
+		jsonRoot := []byte("{}")
+		f, err := os.Create(configPath)
+		if err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Cannot create file '%s': %s", configPath, err))
+		}
+		defer f.Close()
+		_, err = f.Write(jsonRoot)
+		if err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Cannot encode config '%s': %s", configPath, err))
+		}
+	}
+}
+
+func CreateMinishiftDirs(dirs *cmdState.MinishiftDirs) {
+	dirPaths := reflect.ValueOf(*dirs)
+
+	for i := 0; i < dirPaths.NumField(); i++ {
+		path := dirPaths.Field(i).Interface().(string)
+		if err := os.MkdirAll(path, 0777); err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", path))
```

Comment:
too bad this is not atomic, so partial creations can occur.

Reply:
@gbraad I do not think you want me to fix this in this PR as it is out side the scope of this PR

Reply:
out of the scope. At the moment this will not cause issues going forward